core/src/test/scala/ai/h2o/sparkling/backend/api/scalainterpreter/ScalaCodeHandlerSuite.scala

```diff
@@ -181,4 +182,51 @@ class ScalaCodeHandlerSuite extends FunSuite with SharedH2OTestContext with Befo
     assert(!result3.response.equals(""), "Response should not be empty")
   }
 
+  test("Code with exception") {
+    testCode("throw new Exception(\"Exception Message\")", CodeResults.Exception)
+  }
+
+  test("Incomplete code") {
+    testCode("val num = ", CodeResults.Incomplete)
+  }
+
+  test("Simple script which ends successfully") {
+    testCode("val num = 42", CodeResults.Success)
+  }
+
+
+  test("Test successful call after exception occurred") {
+    testCode("throw new Exception(\"Exception Message\")", CodeResults.Exception)
+    testCode("val num = 42", CodeResults.Success)
+  }
+
+  test("Test Spark API call via interpreter") {
+    testCode(
+      """
+        |val list = Seq(('A', 1), ('B', 2), ('A', 3))
+        |val num1 = sc.parallelize(list, 3).groupByKey.count
+        |val num2 = sc.parallelize(list, 3).reduceByKey(_ + _).count
+        |""".stripMargin, CodeResults.Success)
+  }
+
+  test("[SW-386] Test Spark API exposed implicit conversions " +
+    "(https://issues.scala-lang.org/browse/SI-9734 and https://issues.apache.org/jira/browse/SPARK-13456)") {
+    testCode(
+      """
+        |import spark.implicits._
+        |case class Person(id: Long)
+        |val ds = Seq(Person(0), Person(1)).toDS
+        |val count = ds.count
+      """.stripMargin, CodeResults.Success)
+  }
+
+  private def testCode(code: String, expectedResult: CodeResults.Value): Unit = {
```

Comment:
Maybe this method could extend the logic of the `FunSuite.test` function and replace it in the whole suite.

Reply:
👍 Great idea, will do